src/charm.py

```diff
@@ -481,8 +481,7 @@ def _open_ports(self) -> None:
         """
         if ops.JujuVersion.from_environ().supports_open_port_on_k8s:
             try:
-                self.unit.open_port("tcp", 3306)
-                self.unit.open_port("tcp", 33060)
+                self.unit.set_ports(3306, 33060)
```

Comment:
TIL, new for ops 2.7. BTW, did you test it on Juju 2.9? Will it be silently skipped/warned or crashed?